FlowCrypt/Functionality/Services/Compose Message Helper/ComposeMessageHelper.swift

```diff
@@ -210,8 +210,15 @@ final class ComposeMessageHelper {
                 shouldUpdateLastUsed: true
             ).map(\.armored).joined(separator: "\n")
             let parsed = try await core.parseKeys(armoredOrBinary: armoredPubkeys.data())
+            // Fetch public keys which are usableForEncryption
+            // https://github.com/FlowCrypt/flowcrypt-ios/issues/2396
+            // Also fetch revoked or expired keys so that we can display correct error message
+            // UI Test title: sending message to user with expired/revoked public key produces modal
+            let keyDetails = parsed.keyDetails.filter {
+                $0.usableForEncryption || $0.revoked || !$0.isNotExpired
+            }
```

Comment:
probably we should also import sign-only keys to show appropriate error (in the case when there are no valid keys)?
as we already do for revoked and expired keys

Reply:
I think we are seeking to only import correct key (usable for encryption) for message encryption in this issue?

Reply:
No, we should import all keys, but choose only valid ones when encrypting message, or show error message if there are no valid keys for recipient.

In current implementation if user manually imported sign-only key for recipient, then recipient email will be green when composing message, but when you tap `Send` button there will be error message `One or more of your recipients don't have encryption set up`. Which can be confusing for users, as key for recipient was imported and email was green when composing.

So it'll be better to validate keys before encrypting message and show appropriate error message.

Reply:
Could you please explain more about how we should proceed with this user which includes sign-only key?

Recipient status(maybe yellow or green?), what error message should I show when user choose to send message?, do we still have to allow user to send message in this case?

Reply:
Let's say recipient has 2 valid keys and 1 signing-only key.
Does above mean user won't send encrypted message in this case?

Reply:
Also does this mean we won't save draft in this case?

Reply:
> Let's say recipient has 2 valid keys and 1 signing-only key. Does above mean user won't send encrypted message in this case?

No, in this case we should use 2 valid keys and encrypt message with them. So message won't be send only in case when recipient doesn't have any valid key.

> Also does this mean we won't save draft in this case?

We should save draft in this case, but encrypt it only with current user private key (like it's saved when user composed draft but didn't set any recipient)

Reply:
> So we also need to show error message alert only when user doesn't have any valid key(like current sign-only key)

Correct. So if user has any valid key which can be used for encryption - we should use it (or if he has several valid keys, we should use all of them). If no valid key found - then appropriate error message should be shown.

Reply:
Hmm.. white text color doesn't look good on yellow background. Also yellow text color doesn't look good on white background.
So I'm thinking to use dark yellow like below(3rd and 4th image).

![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/a3430d9d-5de6-4f69-8aaf-8ed7637582c2)
![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/a3a48a3f-dd3c-40cc-a560-f6e3ad72937c)
![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/1fe8b4bd-690c-4e8d-baf6-f2d5b57baf7e)
![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/51950a50-6945-40ec-add8-27d995e4fd89)

Reply:
Dark yellow is definitely better 👍